classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php

```diff
@@ -250,6 +250,9 @@ private function _makeRemoteUrlLocalizable() {
 	private function _migrateSubmissionFiles() {
 		import('lib.pkp.classes.submission.SubmissionFile'); // SUBMISSION_FILE_ constants
 
+		// Delete submission_files entries that correspond to nonexistent submissions
+		Capsule::connection()->unprepared('DELETE FROM submission_files WHERE submission_id IN (SELECT sf.submission_id FROM submission_files sf LEFT JOIN submissions s ON sf.submission_id = s.submission_id WHERE s.submission_id IS NULL)');
+
```

Comment:
Do we also need to clean up references to these submission files in other tables? Places I'm thinking of include:

- submission_file_settings
- publication_galleys
- review_files
- review_round_files
- submission_search_objects (`type` = `SUBMISSION_SEARCH_GALLEY_FILE`)

Reply:
Oh I see the `publication_galleys` is handled in each application. :+1: